### PR TITLE
Add the CORS and CSP headers and set proxy if defined

### PR DESCRIPTION
**Pull request by d00rman**
Bug: [T95467](https://phabricator.wikimedia.org/T95467)

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -31,6 +31,31 @@ function initApp(options) {
...
     if(!app.conf.interface) { app.conf.interface = '0.0.0.0'; }
     if(!app.conf.compression_level) { app.conf.compression_level = 3; }
+    if(app.conf.cors === undefined) { app.conf.cors = '*'; }
+    if(!app.conf.csp) {
+        app.conf.csp =
+            "default-src 'self'; object-src 'none'; media-src *; img-src *; style-src *; frame-ancestors 'self'";
+    }
+
+    // set outgoing proxy
+    if(app.conf.proxy) {
```
Comment (gwicke) [inline app.js:41]: http_proxy and https_proxy?

Comment (d00rman) [inline app.js:41]: Yup.

Comment (gwicke) [inline app.js:41]: Sorry for being unclear - I meant: Should we call this http_proxy and/or https_proxy in the config for clarity?

Comment (d00rman) [inline app.js:41]: We already have that as `proxy`, this is a forgotten addition. The basic reason is that in production our proxy handles both, so we don't really need a separation. In fact, I'd argue that separating them makes that less obvious.

Comment (gwicke) [inline app.js:41]: Other protocols can be proxied as well, in which case just 'proxy' would be confusing. I agree that picking one is better than requiring both. Maybe `http_https_proxy`?

**Comment by gwicke** [comment]
Went ahead & merged it. Lets discuss the proxy configuration separately.

**Pull request merged**